tests/unit/steps/test_reset_step.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+
+from repo_smith.steps.reset_step import ResetStep
+
+
+def test_reset_step_parse_with_revision_and_mode():
+    step = ResetStep.parse("n", "d", "id", {"revision": "HEAD~1", "mode": "mixed"})
+    assert isinstance(step, ResetStep)
+    assert step.name == "n"
+    assert step.description == "d"
+    assert step.id == "id"
+    assert step.revision == "HEAD~1"
+    assert step.mode == "mixed"
+    assert step.files is None
+
+
+def test_reset_step_parse_with_files():
+    step = ResetStep.parse(
+        "n",
+        "d",
+        "id",
+        {"revision": "HEAD", "mode": "mixed", "files": ["file1.txt", "file2.txt"]},
+    )
+    assert step.revision == "HEAD"
+    assert step.mode == "mixed"
+    assert step.files == ["file1.txt", "file2.txt"]
+
+
+def test_reset_step_parse_missing_mode():
+    with pytest.raises(ValueError, match='Missing "mode" field in reset step'):
+        ResetStep.parse("n", "d", "id", {"revision": "HEAD~1"})
+
+
+def test_reset_step_parse_missing_revision():
+    with pytest.raises(ValueError, match='Missing "revision" field in reset step'):
+        ResetStep.parse("n", "d", "id", {"mode": "hard"})
+
+
+def test_reset_step_parse_empty_revision():
+    with pytest.raises(ValueError, match='Empty "revision" field in reset step'):
+        ResetStep.parse("n", "d", "id", {"revision": "", "mode": "hard"})
+
+
+def test_reset_step_parse_invalid_mode():
+    with pytest.raises(ValueError, match='Invalid "mode" value'):
```

Comment:
Shouldn't you test if the full string produced is correct? Since you inject a variable for the actual message! I would prefer that